### PR TITLE
update: changed the query for updating total tribe message

### PR DESCRIPTION
**Pull request by tobi-bams**
Updated the query for updating total tribe messages

### CODE REVIEW

**Comment by Evanfeenstra** [comment]
This query needs a `tenant` right? @tobi-bams 
```
const tribes = (await models.Chat.findAll({
        where: { ownerPubkey: contact.publicKey },
 })) as ChatRecord[]
```

Because `Chat` row is created for each tribe member. So for relay in multi-tenant mode, you need ONLY the Chats that owned by a contact on this relay. So add `tenant: contact.id` to the `where` clause

**Comment by tobi-bams** [comment]
> This query needs a `tenant` right? @tobi-bams
> 
> ```
> const tribes = (await models.Chat.findAll({
>         where: { ownerPubkey: contact.publicKey },
>  })) as ChatRecord[]
> ```
> 
> Because `Chat` row is created for each tribe member. So for relay in multi-tenant mode, you need ONLY the Chats that owned by a contact on this relay. So add `tenant: contact.id` to the `where` clause

Yea, that's true.....

Thanks, lemme fix that!!

File: src/utils/setup.ts [hunk 1]
```diff
@@ -157,27 +158,29 @@ const runMigrations = async () => {
...
-      for (let i = 0; i < result.length; i++) {
-        const member = result[i]
-        const totalMessages = await models.Message.count({
-          where: { sender: member.contactId, chatId: member.chatId },
-        })
-        await member.update({ totalMessages })
+    const contacts = (await models.Contact.findAll({
+      where: { isOwner: true },
+    })) as ContactRecord[]
+    for (let i = 0; i < contacts.length; i++) {
```
Comment (kevkevinpal) [inline src/utils/setup.ts:164]: We can use a for each here instead of a c style loop

File: src/utils/setup.ts [hunk 2]
```diff
@@ -157,27 +158,29 @@ const runMigrations = async () => {
...
+      const contact = contacts[i]
+      const tribes = (await models.Chat.findAll({
+        where: { ownerPubkey: contact.publicKey },
+      })) as ChatRecord[]
+      for (let j = 0; j < tribes.length; j++) {
```
Comment (kevkevinpal) [inline src/utils/setup.ts:169]: Same here use a for each instead of c style loop

File: src/utils/setup.ts [hunk 3]
```diff
@@ -157,27 +158,29 @@ const runMigrations = async () => {
...
+        const tribe = tribes[j]
+        const tribeMembers = (await models.ChatMember.findAll({
+          where: { chatId: tribe.id },
+        })) as ChatMemberRecord[]
+        for (let k = 0; k < tribeMembers.length; k++) {
```
Comment (kevkevinpal) [inline src/utils/setup.ts:174]: Last one use for each instead of c style loop

**Review by kevkevinpal — COMMENTED** [review]
For each instead of c style looping

File: src/utils/setup.ts (hunk 3 above)
Comment (Evanfeenstra) [reply src/utils/setup.ts:174]: disagree... for loop is "async aware"... forEach will be faster but could also overflow the SQLITE connection and cause a crash with many members (all queries would be running simultaneously)

**Pull request merged**